x-pack/plugin/logsdb/src/test/java/org/elasticsearch/xpack/logsdb/SyntheticSourceLicenseServiceTests.java

```diff
@@ -8,54 +8,193 @@
 package org.elasticsearch.xpack.logsdb;
 
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.license.License;
+import org.elasticsearch.license.LicenseService;
 import org.elasticsearch.license.MockLicenseState;
+import org.elasticsearch.license.TestUtils;
 import org.elasticsearch.test.ESTestCase;
+import org.junit.Before;
 import org.mockito.Mockito;
 
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.UUID;
+
+import static org.elasticsearch.license.TestUtils.dateMath;
 import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.same;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
 public class SyntheticSourceLicenseServiceTests extends ESTestCase {
 
+    private LicenseService mockLicenseService;
+    private SyntheticSourceLicenseService licenseService;
+
+    @Before
+    public void setup() throws Exception {
+        mockLicenseService = mock(LicenseService.class);
+        License license = createEnterpirseLicense();
+        when(mockLicenseService.getLicense()).thenReturn(license);
+        licenseService = new SyntheticSourceLicenseService(Settings.EMPTY);
+    }
+
     public void testLicenseAllowsSyntheticSource() {
         MockLicenseState licenseState = mock(MockLicenseState.class);
```

Comment:
```suggestion
        MockLicenseState licenseState = MockLicenseState.createMock();
```
here and throughout -- that way `copyCurrentLicenseState` (see my other suggestion in prod-code) will work as expected

Reply:
Or actually, instead of the mock, I think using the real class makes for better coverage:

`XPackLicenseState licenseState = new XPackLicenseState(() -> 1, new XPackLicenseStatus(license.operationMode(), true, null));` creates the license state, and then usage can be verified with 

`assertThat(licenseState.getLastUsed(), aMapWithSize(1));`

Just a suggestion -- feel free to reject.

Reply:
I will keep using the mocks for now. I don't think this is too bad.